app/src/test/java/org/simple/clinic/util/identifierdisplay/IdentifierDisplayAdapterTest.kt

```diff
@@ -91,11 +93,81 @@ class IdentifierDisplayAdapterTest {
     )
   }
 
+  @Test
+  fun `the adapter should use the provided converter for converting the identifier type`() {
+    val identifierDisplayAdapter = IdentifierDisplayAdapter(
+        converters = mapOf(
+            BpPassport to StubIdentifierToTextConverter(convertTypeAction = { "bp_passport" })
+        ),
+        unknownValueFallback = { throw RuntimeException() },
+        unknownTypeFallback = { throw RuntimeException() }
+    )
+
+    val convertedForDisplay = identifierDisplayAdapter.typeAsText(Identifier(value = "123", type = BpPassport))
+
+    assertThat(convertedForDisplay).isEqualTo("bp_passport")
+  }
+
+  @Test
+  @Parameters(method = "params for unknown identifier type fallback")
+  fun `types of unknown identifiers should be converted to using the fallback`(
+      fallback: (Identifier) -> String,
+      unknownIdentifier: Identifier,
+      expectedConvertedUnknownType: String
+  ) {
+    val identifierDisplayAdapter = IdentifierDisplayAdapter(
+        converters = mapOf(
+            BpPassport to StubIdentifierToTextConverter(convertTypeAction = { "bp_passport" })
+        ),
+        unknownValueFallback = { throw RuntimeException() },
+        unknownTypeFallback = fallback
+    )
+    val knownIdentifier = Identifier(value = "id_1", type = BpPassport)
+
+    val convertedForDisplay = listOf(knownIdentifier, unknownIdentifier)
+        .map(identifierDisplayAdapter::typeAsText)
+
+    val expected = listOf("bp_passport", expectedConvertedUnknownType)
+    assertThat(convertedForDisplay).isEqualTo(expected)
+  }
+
+  @Suppress("Unused")
+  private fun `params for unknown identifier type fallback`(): List<List<Any>> {
```

Comment:
The fallback for all `Unknown` identifiers will always be the same.
Hence, I think, keeping the `fallback` configurable is not adding any value.

To use `UnknownIdentifierFormatter` also makes sense for testing and we don't need to do parametrized testing for different fallbacks for `Unknown` anymore.

Reply:
I've done this in the last commit. Please check?